docs/kubernetes/deploy.md

```diff
@@ -99,7 +99,7 @@ If you don't have an SSH key [cluster operators may generate a new one](../ssh.m
 
 ### Step 2: Create a Service Principal
 
-Kubernetes clusters have integrated support for various cloud providers as core functionality. On Azure, acs-engine uses a Service Principal to interact with Azure Resource Manager (ARM). Follow the instructions to [create a new service principal](../serviceprincipal.md)
+Kubernetes clusters have integrated support for various cloud providers as core functionality. On Azure, acs-engine uses a Service Principal to interact with Azure Resource Manager (ARM). Follow the instructions to [create a new service principal and grant it the necessary IAM role to create Azure resources](../serviceprincipal.md).
```

Comment:
This is a good addition, but having a URL link that long looks strange. Could you add the words outside the URL link markdown?